keras_cv/layers/preprocessing/vectorized_base_image_augmentation_layer.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.utils import preprocessing
+
+# In order to support both unbatched and batched inputs, the horizontal
+# and verticle axis is reverse indexed
+H_AXIS = -3
+W_AXIS = -2
+
+IMAGES = "images"
+LABELS = "labels"
+TARGETS = "targets"
+BOUNDING_BOXES = "bounding_boxes"
+KEYPOINTS = "keypoints"
+SEGMENTATION_MASKS = "segmentation_masks"
+
+IS_DICT = "is_dict"
+BATCHED = "batched"
+USE_TARGETS = "use_targets"
+
+
+class VectorizedBaseImageAugmentationLayer(
```

Comment:
Is there a reason this can't subclass BaseImageAugmentationLayer?

Reply:
discussed a bit offline but to summarize for community members:

they barely share functionality in reality despite looking like they may.